internal/ad/ad.go

```diff
@@ -389,6 +409,57 @@ func (ad *AD) ensureKrb5CCName(srcKrb5CCName, dstKrb5CCName string) (err error)
 	return nil
 }
 
+// ensureKrb5CCCopy checks if the cached krb5cc ticket is up to date and updates
+// it if not.
+func ensureKrb5CCCopy(srcKrb5CCName, krb5CCCopyName string) (err error) {
+	if copyStat, err := os.Stat(krb5CCCopyName); err == nil {
+		// We already have a copy of the ticket, let's check if we need to update it
+		srcStat, err := os.Stat(srcKrb5CCName)
+		if err != nil {
+			return fmt.Errorf(i18n.G("failed to stat source ticket: %w"), err)
+		}
+
+		// The source ticket is not newer than the destination one, no need to update
+		if !srcStat.ModTime().After(copyStat.ModTime()) {
+			return nil
+		}
+	}
+
+	// The ticket is either not present or outdated, let's update it
+	// Create the parent directory if it doesn't exist
+	if err := os.MkdirAll(filepath.Dir(krb5CCCopyName), 0700); err != nil {
+		return fmt.Errorf(i18n.G("failed to create parent directory for ticket copy: %w"), err)
+	}
+
+	// Copy the ticket
+	if err := copyFile(srcKrb5CCName, krb5CCCopyName, 0600); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// copyFile copies a file from src to dst, using the specified mode.
+func copyFile(src, dst string, mode os.FileMode) error {
+	srcFile, err := os.Open(src)
+	if err != nil {
+		return fmt.Errorf(i18n.G("failed to open source file: %w"), err)
+	}
+	defer srcFile.Close()
+
+	dstFile, err := os.OpenFile(dst, os.O_CREATE|os.O_WRONLY|os.O_TRUNC, mode)
+	if err != nil {
+		return fmt.Errorf(i18n.G("failed to open destination file: %w"), err)
+	}
+	defer dstFile.Close()
+
+	if _, err := io.Copy(dstFile, srcFile); err != nil {
+		return fmt.Errorf(i18n.G("failed to copy file: %w"), err)
+	}
+
+	return nil
+}
```

Comment:
Instead of having two file descriptors opened and potentially opening ourselves for tragedy, can't we read from `src` and then safely write (i.e. writing as .new and then renaming it) to `dest`?

Reply:
I was thinking more in the way of reading the file first with `os.ReadFile` and then writing it and renaming with `os.WriteFile` and `os.Rename`. This way we can avoid the `io` package alltogether.